File: cera-ffi/examples/ExplicitLoading.kt
import uniffi.cera_ffi.BackendPreference
import uniffi.cera_ffi.EngineConfig
import uniffi.cera_ffi.GenerateOpts
import uniffi.cera_ffi.ModelLoader
import uniffi.cera_ffi.ModelSource
import uniffi.cera_ffi.SessionConfig

// Run with a generative GGUF path and a raw completion prompt.
fun main(args: Array<String>) {
    require(args.size == 2) { "usage: ExplicitLoadingKt model.gguf prompt" }
    ModelLoader(ModelSource.Path(args[0]), EngineConfig(backend = BackendPreference.CPU)).use { loader ->
        loader.buildGenerative().use { model ->
            model.engine().use { engine ->
                model.createSession(SessionConfig(seed = 42uL)).use { session ->
                    session.appendTokens(engine.encodeText(args[1]))
                    val output = session.generate(GenerateOpts(maxTokens = 32u, temperature = 0.7f))
                    println(engine.decodeTokens(output.tokens))
                }
            }
        }
    }
}

File: cera-ffi/examples/ExplicitLoading.swift
import Cera
import Foundation

// Run with a generative GGUF path and a raw completion prompt.
@main
struct ExplicitLoading {
  static func main() throws {
    guard CommandLine.arguments.count == 3 else {
      throw NSError(
        domain: "ExplicitLoading", code: 1,
        userInfo: [NSLocalizedDescriptionKey: "usage: explicit-loading model.gguf prompt"])
    }
    let loader = ModelLoader(
      source: .path(path: CommandLine.arguments[1]), config: EngineConfig(backend: .cpu))
    let model = try loader.buildGenerative()
    let engine = model.engine()
    let session = try model.createSession(config: SessionConfig(seed: 42))
    let prompt = engine.encodeText(text: CommandLine.arguments[2])
    try session.appendTokens(tokens: prompt)
    let output = try session.generate(opts: GenerateOpts(maxTokens: 32, temperature: 0.7))
    print(engine.decodeTokens(tokens: output.tokens))
  }
}
